nifi-nar-bundles/nifi-workday-bundle/nifi-workday-processors/src/main/java/org/apache/nifi/processors/workday/GetWorkdayReport.java

```diff
@@ -0,0 +1,432 @@
+        int numberOfRecords = 0;
+        String mimeType = null;
+        try (RecordReader reader = recordReaderFactoryAtomicReference.get().createRecordReader(flowfile,
+            new BufferedInputStream(responseBodyStream), getLogger())) {
+            RecordSchema schema = recordSetWriterFactoryAtomicReference.get()
+                .getSchema(flowfile == null ? Collections.emptyMap() : flowfile.getAttributes(), reader.getSchema());
+            try (OutputStream responseStream = session.write(responseFlowFile);
+                RecordSetWriter recordSetWriter = recordSetWriterFactoryAtomicReference.get().createWriter(getLogger(), schema, responseStream, responseFlowFile)) {
+                mimeType = recordSetWriter.getMimeType();
+                recordSetWriter.beginRecordSet();
+                Record currentRecord;
+                while ((currentRecord = reader.nextRecord(false, true)) != null) {
+                    for (String recordPath : columnsToHash) {
+                        RecordPathResult evaluate = RecordPath.compile("hash(" + recordPath + ", '" + hashingAlgorithm + "')").evaluate(currentRecord);
+                        evaluate.getSelectedFields().forEach(fieldVal -> fieldVal.updateValue(fieldVal.getValue(), RecordFieldType.STRING.getDataType()));
+                    }
+                    currentRecord.incorporateInactiveFields();
+                    recordSetWriter.write(currentRecord);
+                    numberOfRecords++;
+                }
+            }
+        }
+        return new TransformResult(numberOfRecords, mimeType);
+    }
+
+    private void route(FlowFile request, FlowFile response, ProcessSession session, ProcessContext context, int statusCode) {
+        if (!isSuccess(statusCode) && request == null) {
+            context.yield();
+        }
+
+        if (isSuccess(statusCode)) {
+            if (request != null) {
+                session.transfer(request, ORIGINAL);
+            }
+            if (response != null) {
+                session.transfer(response, RESPONSE);
+            }
+        } else {
+            if (request != null) {
+                session.transfer(request, FAILURE);
+            }
+        }
+    }
+
+    private boolean isSuccess(int statusCode) {
+        return statusCode / 100 == 2;
+    }
+
+    private Map<String, String> createCommonAttributes(URI uri, HttpResponseEntity httpResponseEntity, long elapsedTime) {
+        Map<String, String> attributes = new HashMap<>();
+        attributes.put(STATUS_CODE, String.valueOf(httpResponseEntity.statusCode()));
+        attributes.put(REQUEST_URL, uri.toString());
+        attributes.put(REQUEST_DURATION, Long.toString(elapsedTime));
+        attributes.put(TRANSACTION_ID, UUID.randomUUID().toString());
+        return attributes;
+    }
+
+    private Map<String, String> decorateWithMimeAttribute(Map<String, String> commonAttributes, HttpResponseEntity httpResponseEntity) {
```

Comment:
Recommend renaming this method:
```suggestion
    private Map<String, String> setMimeType(Map<String, String> commonAttributes, HttpResponseEntity httpResponseEntity) {
```

Reply:
done